setup.py

```diff
@@ -120,12 +120,10 @@
             # Type check
             "pytype",
             "mypy",
-            # Lint code (flake8 replacement)
-            "ruff",
-            # Sort imports
-            "isort>=5.0",
+            # Lint code and sort imports (flake8 and isort replacement)
+            "ruff>=0.0.288",
             # Reformat
-            "black",
+            "black>=23.9.1,<24",
```

Comment:
Btw `black~=23.9` should do the same here

Reply:
yes it is, I do find `>=23.9.1,<24` more explicit though (the thing is `~=` is not consistent among package managers/languages... :/ see https://jubianchi.github.io/semver-check/#/~23.9/23.10 for instance).

Reply:
Oh I did not know this! Is this also the case in the python world?